### PR TITLE
feat: button to delete all downloads

### PR DESCRIPTION
**Pull request by thevedantchourey**
Button to delete all downloaded videos at once #5526
I have added delete button to delete all downloaded videos at once.

### CODE REVIEW

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt [hunk 1]
```diff
@@ -168,12 +212,13 @@ class DownloadsFragment : DynamicLayoutManagerFragment() {
         super.onStart()
     }
 
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt:215]: That makes the fragment crash or behave wrong on older Android versions ...

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt [hunk 2]
```diff
@@ -155,11 +158,52 @@ class DownloadsFragment : DynamicLayoutManagerFragment() {
...
+
+
         binding.shuffleBackground.setOnClickListener {
             BackgroundHelper.playOnBackgroundOffline(requireContext(), null)
         }
     }
 
+    private fun showDeleteDialog(context: Context) {
+        MaterialAlertDialogBuilder(context)
+            .setTitle("All downloads will be deleted!")
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt:177]: Please make that translatable.

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt [hunk 3]
```diff
@@ -155,11 +158,52 @@ class DownloadsFragment : DynamicLayoutManagerFragment() {
...
+            }
+        }
+
+
         binding.shuffleBackground.setOnClickListener {
             BackgroundHelper.playOnBackgroundOffline(requireContext(), null)
         }
     }
 
+    private fun showDeleteDialog(context: Context) {
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt:175]:
```suggestion
    private fun showDeleteAllDialog(context: Context) {
```

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt [hunk 4]
```diff
@@ -155,11 +158,52 @@ class DownloadsFragment : DynamicLayoutManagerFragment() {
             }
         )
 
+
+        if (dbDownloads.isNotEmpty()){
+            binding.deleteAll.visibility = View.VISIBLE
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt:163]:
```suggestion
            binding.deleteAll.isVisible = true
```

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt [hunk 5]
```diff
@@ -83,6 +86,7 @@ class DownloadsFragment : DynamicLayoutManagerFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        binding.deleteAll.visibility = View.INVISIBLE
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt:89]:
```suggestion
        binding.deleteAll.isInvisible = true
```

File: app/src/main/java/com/github/libretube/ui/adapters/DownloadsAdapter.kt
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/adapters/DownloadsAdapter.kt:1]: Please undo changes to this file.

Comment (thevedantchourey) [reply app/src/main/java/com/github/libretube/ui/adapters/DownloadsAdapter.kt:1]: I do not change anything in download adapter file.

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt (hunk 1 above)
Comment (thevedantchourey) [reply app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt]: I have undone that change.

File: app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt (hunk 2 above)
Comment (thevedantchourey) [reply app/src/main/java/com/github/libretube/ui/fragments/DownloadsFragment.kt]: i have added that to string.

**Review by Bnyro — APPROVED** [review]
Thanks!

**Comment by thevedantchourey** [comment]
> Thanks!
Pleasure is all mine.

**Comment by thevedantchourey** [comment]
Will you include my name in the list of contributors for the next release?

**Pull request merged**